src/lib.rs

```diff
@@ -667,16 +665,54 @@ impl Config {
                     }
                     cmd.arg("-AWin32");
                 } else {
-                    panic!("unsupported msvc target: {}", target);
+                    panic!("unsupported msvc target: {target}");
                 }
             }
-        } else if target.contains("darwin") && !self.defined("CMAKE_OSX_ARCHITECTURES") {
-            if target.contains("x86_64") {
-                cmd.arg("-DCMAKE_OSX_ARCHITECTURES=x86_64");
-            } else if target.contains("aarch64") {
-                cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64");
-            } else {
-                panic!("unsupported darwin target: {}", target);
+        } else if target.contains("apple") {
+            if !self.defined("CMAKE_OSX_ARCHITECTURES") {
+                if target.contains("x86_64") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=x86_64");
+                } else if target.contains("aarch64") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64");
+                } else if target.contains("i686") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=i686");
+                } else if target.contains("arm64e") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64e");
+                } else if target.contains("i386") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=i386");
+                } else if target.contains("armv7s") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=armv7s");
+                } else if target.contains("armv7k") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=armv7k");
+                } else if target.contains("arm64_32") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64_32");
+                } else {
+                    panic!("unsupported apple target: {target}");
+                }
+            }
+
+            if !self.defined("CMAKE_OSX_SYSROOT") {
```

Comment:
Reading the docs for this, it seems to me that we can (and should) use SDKROOT instead (to allow the user to override with that env var set). And `cc-rs` already contains the logic to invoke `xcrun` at the appropriate times.

So maybe the solution here is to instead make `cc-rs` set the `SDKROOT` env var [here](https://github.com/rust-lang/cc-rs/blob/d61b3b3f393ad20c41849ad004da153e527b0503/src/lib.rs#L2743-L2744) instead of passing `-isysroot`? That would automatically set `SDKROOT` for CMake invocations as well.

Reply:
Sounds good, I'll try making the proposed change in `cc-rs` and see how that affects `cmake` behaviour.

Reply:
Seems like we don't need any of this extra code if `SDKROOT` is set, so I'll PR that upstream.